src/trace-server.ts

```diff
@@ -168,27 +168,57 @@ export class TraceServer {
             },
             async progress => {
                 progress.report({ message: 'starting up...' });
-                let timeout = false;
-                const timeoutId = setTimeout(() => (timeout = true), millis);
-
-                // eslint-disable-next-line no-constant-condition
-                while (true) {
-                    if (await this.isUp()) {
-                        this.showStatus(true);
-                        clearTimeout(timeoutId);
-                        break;
-                    }
-                    if (timeout) {
-                        this.showError(prefix + ' startup timed-out after ' + millis + 'ms.');
-                        await this.stopOrReset(context);
-                        break;
+                try {
+                    let timeout = false;
+                    const timeoutId = setTimeout(() => (timeout = true), millis);
+
+                    let stderr = '';
+                    this.server?.stderr?.on('data', data => {
+                        stderr = data;
+                    });
+
+                    let errorString: string | undefined;
+                    this.server?.once('exit', code => {
+                        errorString = 'Code: ' + code + (stderr !== '' ? '\nstdErr: ' + stderr : '');
+                    });
+
+                    this.server?.once('error', () => {
+                        errorString = 'Error occurred.' + (stderr !== '' ? '\nstdErr: ' + stderr : '');
```

Comment:
After further research / debugging, I am no longer sure if this listener is useful. It seems the most likely cases where an "error" event would be emitted, execution will not get this far in the code? From what I see such error are caught in the "start" method, when it's noticed that `server.pid` is not defined:

![image](https://github.com/user-attachments/assets/8ab97abe-7ba3-4baa-94de-1d292be81359)

If we can safely remove this listener, there should be no listener left to clean-up upon exit and no need for a "finally" block?

Reply:
Side-note: in the case mentioned above, the error message could be more detailed. e.g. I changed the trace server path so it's wrong, and I get:
![image](https://github.com/user-attachments/assets/da94da17-72a7-4eed-8a53-71196377f69c)

Reply:
Error listener removed and startup errors are handled in newest version